File: FirmwareGO/application/services/ApplicationStateService.go
package services

import (
	. "FirmwareGO/domain/entities"
	. "FirmwareGO/infrastructure/repositories/interfaces"
)

type ApplicationStateService struct {
	Repo IRepository[*BaseEntity]
}

func NewApplicationStateService(repo IRepository[*BaseEntity]) *ApplicationStateService {
	return &ApplicationStateService{Repo: repo}
}

func (s *ApplicationStateService) DbIsConnected() bool {
	return s.Repo.IsDatabaseConnected()
}

File: NotificationMS/Application/ApplicationServices/ApplicationStateService.cs
using Application.ApplicationServices.Interfaces;
using Domain.Entities;
using Infrastructure.Repositories.Interfaces;

namespace Application.ApplicationServices;

public class ApplicationStateService : IApplicationStateService
{
    private readonly IRepository<BaseEntity> _repository;

    public ApplicationStateService(IRepository<BaseEntity> repository)
    {
        _repository = repository;
    }

    public async Task<bool> DbIsConnected()
    {
        return await _repository.IsDatabaseConnected();
    }
}